cardano_node_tests/tests/test_plutus_spend_build.py

```diff
@@ -1209,3 +1238,56 @@ def test_collateral_is_txin(
 
         dbsync_utils.check_tx(cluster_obj=cluster, tx_raw_output=tx_output_step1)
         dbsync_utils.check_tx(cluster_obj=cluster, tx_raw_output=tx_output_step2)
+
+
+@pytest.mark.testnets
+class TestNegative:
+    """Tests for txin locking using Plutus smart contracts that are expected to fail."""
+
+    @allure.link(helpers.get_vcs_link())
+    @pytest.mark.testnets
+    @hypothesis.given(redeemer_number=st.integers(min_value=-2 ^ 64, max_value=2 ^ 64))
+    @common.hypothesis_settings()
+    def test_guessing_game_pbt(
+        self,
+        cluster_lock_guessing_game: clusterlib.ClusterLib,
+        payment_addrs_lock_guessing_game: List[clusterlib.AddressRecord],
+        fund_script_guessing_game: clusterlib.TxRawOutput,
+        redeemer_number: int,
+    ):
+        """Try to spend a locked UtxO with and an invalid redeemer.
+
+        Expect failure.
+        """
+        hypothesis.assume(redeemer_number != 42)
+
+        cluster = cluster_lock_guessing_game
+        temp_template = f"{common.get_test_id(cluster)}"
+
+        redeemer_file = "redeemer_file.datum"
+        with open(redeemer_file, "w", encoding="utf-8") as outfile:
+            json.dump({"constructor": 0, "fields": [{"int": redeemer_number}]}, outfile)
+
+        plutus_op = plutus_common.PlutusOp(
+            script_file=plutus_common.GUESSING_GAME_PLUTUS,
+            datum_file=plutus_common.DATUM_42_TYPED,
+            redeemer_file=Path(redeemer_file),
+        )
+
+        txid = cluster.get_txid(tx_body_file=fund_script_guessing_game.out_file)
+        script_utxos = cluster.get_utxo(txin=f"{txid}#1")
+        collateral_utxos = cluster.get_utxo(txin=f"{txid}#2")
+
+        with pytest.raises(clusterlib.CLIError) as excinfo:
+            _build_spend_locked_txin(
+                temp_template=temp_template,
+                cluster_obj=cluster,
+                payment_addr=payment_addrs_lock_guessing_game[0],
+                dst_addr=payment_addrs_lock_guessing_game[1],
+                script_utxos=script_utxos,
+                collateral_utxos=collateral_utxos,
+                plutus_op=plutus_op,
+                amount=50_000_000,
+            )
+
+        assert "The Plutus script evaluation failed" in str(excinfo.value)
```

Comment:
Can the expected error be made more specific, so we know it is failing because of redeemer value?

Reply:
```
Command failed: transaction build  Error: The following scripts have execution failures:
the script for transaction input 0 (in the order of the TxIds) failed with: 
The Plutus script evaluation failed: An error has occurred:  User error:
The provided Plutus code called 'error'.
Script debugging logs: Pd
```
I am getting this error message, non specific information